lib/pbench/server/api/auth.py

```diff
@@ -0,0 +1,127 @@
+import jwt
+import os
+import datetime
+from flask import request, abort
+from flask_httpauth import HTTPTokenAuth
+from pbench.server.database.models.models import UserModel, ActiveTokens
+from pbench.server.database.database import Database
+
+
+class Auth:
+    token_auth = HTTPTokenAuth("Bearer")
+
+    @staticmethod
+    def set_logger(logger):
+        # Logger gets set at the time of auth module initialization
+        Auth.logger = logger
+
+    def encode_auth_token(self, token_expire_duration, user_id):
+        """
+        Generates the Auth Token
+        :return: string
+        """
+        payload = {
+            "iat": datetime.datetime.utcnow(),
+            "exp": datetime.datetime.utcnow()
```

Comment:
Seems like we should fetch the time once, and use that value as the `iat` value, and then calculate the `exp` value with it.

Reply:
fixed